src/annotator/plugin/bucket-bar.js

```diff
@@ -108,101 +78,17 @@ export default class BucketBar extends Delegator {
   }
 
   _update() {
-    this.buckets = anchorBuckets(this.annotator.anchors);
-
-    // The following affordances attempt to reuse existing DOM elements
-    // when reconstructing bucket "tabs" to cut down on the number of elements
-    // created and added to the DOM
-
-    // Only leave as many "tab" elements attached to the DOM as there are
-    // buckets
-    this.tabs.slice(this.buckets.length).forEach(tabEl => tabEl.remove());
-
-    // And cut the "tabs" collection down to the size of buckets, too
-    /** @type {HTMLElement[]} */
-    this.tabs = this.tabs.slice(0, this.buckets.length);
-
-    // If the number of "tabs" currently in the DOM is too small (fewer than
-    // buckets), fill that gap by creating new elements (and adding event
-    // listeners to them)
-    this.buckets.slice(this.tabs.length).forEach(() => {
-      const tabEl = document.createElement('div');
-      this.tabs.push(tabEl);
-
-      // Note that these elements are reused as buckets change, meaning that
-      // any given tab element will correspond to a different bucket over time.
-      // However, we know that we have one "tab" per bucket, in order,
-      // so we can look up the correct bucket for a tab at event time.
-
-      // Focus and unfocus highlights on mouse events
-      tabEl.addEventListener('mousemove', () => {
-        this.updateHighlightFocus(this.tabs.indexOf(tabEl), true);
-      });
-
-      tabEl.addEventListener('mouseout', () => {
-        this.updateHighlightFocus(this.tabs.indexOf(tabEl), false);
-      });
-
-      // Select the annotations (in the sidebar)
-      // that have anchors within the clicked bucket
-      tabEl.addEventListener('click', event => {
-        event.stopPropagation();
-        const index = this.tabs.indexOf(tabEl);
-        const bucket = this.buckets[index];
-        if (!bucket) {
-          return;
+    const buckets = anchorBuckets(this.annotator.anchors);
```

Comment:
These changes do away with the reuse of DOM elements and whatnot. I think preact is likely smarter here than we ever could be manually.